rero_ils/modules/notifications/dispatcher.py

```diff
@@ -19,127 +19,260 @@
 
 from __future__ import absolute_import, print_function
 
+import pycountry
 from flask import current_app
 from invenio_mail.api import TemplatedMessage
 from invenio_mail.tasks import send_email as task_send_email
+from num2words import num2words
 
-from ..locations.api import Location
+from ..libraries.api import email_notification_type
 
 
 class Dispatcher:
     """Dispatcher notifications class."""
 
-    def dispatch_notification(self, notification=None, verbose=False):
-        """Dispatch the notification."""
-        from .api import get_communication_channel_to_use
+    @classmethod
+    def dispatch_notifications(cls, notification_pids=[], resend=False,
+                               verbose=False):
+        """Dispatch the notification.
+
+        :param notification_pids: Notification pids to send notification.
+        :param resend: Resend notification if already send.
+        :param verbose: Verbose output.
+        :returns: ictionair with proccessed and send count
+        """
+        from .api import Notification, get_communication_channel_to_use, \
+            get_template_to_use
 
         def not_yet_implemented(*args):
             """Do nothing placeholder for a notification."""
             return
 
-        if notification:
-            data = notification.replace_pids_and_refs()
-            communication_switcher = {
-                'email': Dispatcher.send_mail_to_patron,
-                'mail': Dispatcher.send_mail_for_printing,
-                #  'sms': not_yet_implemented
-                #  'telepathy': self.madness_mind
-                #  ...
-            }
-            patron = data['loan']['patron']
-            dispatcher_function = communication_switcher.get(
-                get_communication_channel_to_use(
-                    notification.init_loan(), notification, patron
-                ),
-                not_yet_implemented
-            )
-            if dispatcher_function == not_yet_implemented:
-                current_app.logger.warning(
-                    'The communication channel of the patron (pid: {pid})'
-                    'is not yet implemented'.format(pid=patron['pid'])
+        send = 0
+        not_send = 0
+        count = 0
```

Comment:
COSMETIC : you could merge into a single line : `send = not_send = count = 0`
All variables keeps  its independence